packages/abi-coder/src/encoding/coders/EnumCoder.ts

```diff
@@ -80,7 +80,7 @@ export class EnumCoder<TCoders extends Record<string, Coder>> extends Coder<
   }
 
   decode(data: Uint8Array, offset: number): [DecodedValueOf<TCoders>, number] {
-    if (this.#shouldValidateLength && data.length < this.#encodedValueSize) {
+    if (this.#shouldValidateLength && data.length < this.encodedLength) {
```

Comment:
Here we were not including the case length in the calculation. Meaning `#encodedValueSize` was `1` if the value was `u8`. An enum is always at least `WORD_SIZE` due to the case.